### PR TITLE
[PF-562] renaming pass

### PR DESCRIPTION
**Pull request by ddietterich**
This PR is exclusively renames. The changes are:
- Add an `Api` prefix to all generated classes. Now it is obvious where they are being used in the code.
- Rename Google, GOOGLE, google to Gcp, GCP, and gcp. The exception is that I used `gcsBucket`
- Rename all things "reference" to "referenced"
- Rename in the API the schema CloudContext to CloudPlatform
- Rename the ReferenceController to ReferencedGcpResourcesController

So it is a mess of files, but only renamings.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -149,6 +149,7 @@ swaggerSources {
 			rawOptions = [
 					"--api-package", "${artifactGroup}.generated.controller",
 					"--model-package", "${artifactGroup}.generated.model",
+					"--model-name-prefix", "Api",
```
Comment (jaycarlton) [inline build.gradle:152]: Did you do the trick where you refactor the generated class in IJ first to fix up all the references to it and then generate them with the prefix?

File: src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java [hunk 2]
```diff
@@ -283,120 +288,112 @@ private String getAsyncResultEndpoint(String jobId) {
...
+  private ApiCreateCloudContextResult fetchCreateCloudContextResult(
       String jobId, AuthenticatedUserRequest userReq) {
     final AsyncJobResult<GcpCloudContext> jobResult =
         jobService.retrieveAsyncJobResult(jobId, GcpCloudContext.class, userReq);
 
-    final GcpContext gcpContext;
-    if (jobResult.getJobReport().getStatus().equals(StatusEnum.SUCCEEDED)) {
-      gcpContext = new GcpContext().projectId(jobResult.getResult().getGcpProjectId());
+    final ApiGcpContext gcpContext;
+    if (jobResult.getApiJobReport().getStatus().equals(StatusEnum.SUCCEEDED)) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:382]: I don't know if you wanted names inside functions to be changed or not.

File: src/main/java/bio/terra/workspace/common/utils/ControllerValidationUtils.java [hunk 3]
```diff
@@ -75,10 +75,10 @@ public static void validateEmail(String email) {
   }
 
   /** Validate that a user is requesting a valid cloud for adding workspace context. */
-  public static void validateCloudPlatform(CloudContext context) {
-    if (context != CloudContext.GOOGLE) {
+  public static void validateCloudPlatform(ApiCloudPlatform platform) {
+    if (platform != ApiCloudPlatform.GCP) {
       throw new AzureNotImplementedException(
-          "Invalid cloud context. Currently, only Google contexts are supported.");
+          "Invalid cloud platform. Currently, only GCP is are supported.");
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/common/utils/ControllerValidationUtils.java:81]: "is are"

File: src/main/java/bio/terra/workspace/service/iam/model/IamRole.java [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
+import bio.terra.workspace.generated.model.ApiIamRole;
 import java.util.Arrays;
 import java.util.Optional;
 
 /** Internal representation of IAM roles. */
 public enum IamRole {
-  READER("reader", bio.terra.workspace.generated.model.IamRole.READER),
-  WRITER("writer", bio.terra.workspace.generated.model.IamRole.WRITER),
-  OWNER("owner", bio.terra.workspace.generated.model.IamRole.OWNER);
+  READER("reader", bio.terra.workspace.generated.model.ApiIamRole.READER),
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/service/iam/model/IamRole.java:9]: You don't need the qualification anymore here.

File: src/main/resources/api/service_openapi.yaml [hunk 5]
```diff
@@ -238,7 +239,7 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
-  /api/workspaces/v1/{workspaceId}/resources/references/datarepo/snapshots/{referenceId}:
+  /api/workspaces/v1/{workspaceId}/resources/referenced/datarepo/snapshots/{referenceId}:
```
Comment (jaycarlton) [inline src/main/resources/api/service_openapi.yaml:242]: excellent, you took the bait

File: src/main/resources/api/service_openapi.yaml [hunk 6]
```diff
@@ -768,12 +771,112 @@ components:
...
+    # TODO(PF-404): remove this in favor of reference-type specific objects.
+    CreateDataReferenceRequestBody:
+      type: object
+      required: [name, cloningInstructions]
+      properties:
+        name:
+          $ref: "#/components/schemas/Name"
+        description:
+          type: string
+          default: ""
```
Comment (jaycarlton) [inline src/main/resources/api/service_openapi.yaml:790]: Do we prefer empty strings to null?

File: src/main/resources/api/service_openapi.yaml [hunk 7]
```diff
@@ -768,12 +771,112 @@ components:
...
+      description: A cloud platform which holds context for a workspace.
       required: true
       schema:
-        $ref: '#/components/schemas/CloudContext'
+        $ref: '#/components/schemas/CloudPlatform'
 
   schemas:
+    ### DEPRECATED SCHEMAS ###
+    # TODO(PF-404): remove this in favor of reference-type specific objects.
+    CreateDataReferenceRequestBody:
```
Comment (jaycarlton) [inline src/main/resources/api/service_openapi.yaml:782]: Maybe rename to `DeprecatedCreateDataReferenceRequestBody` here and below.

File: src/main/resources/api/service_openapi.yaml [hunk 8]
```diff
@@ -937,29 +1040,20 @@ components:
...
+      enum: ['GCP']
 
     CreateCloudContextRequest:
       type: object
-      required: [cloudType, jobControl]
+      required: [cloudPlatform, jobControl]
       description: |
         Request body for asynchronously creating a cloud context for a workspace.
-        Contains the requested CloudContext enum and JobControl object.
+        Contains the CloudPlatform for the context and and the JobControl object.
```
Comment (jaycarlton) [inline src/main/resources/api/service_openapi.yaml:1053]: "and and"

File: src/main/resources/api/service_openapi.yaml [hunk 9]
```diff
@@ -1306,18 +1316,18 @@ components:
         bucketName:
           description: Name of created bucket (not the resource nmae).
           type: string
-    GoogleBucketDefaultStorageClass:
+    GcsBucketDefaultStorageClass:
       description: >-
         Type and availability of objects in a bucket, described at https://cloud.google.com/storage/docs/storage-classes.
         This list does not include legacy types that apply to some existing buckets. For those, see
-        GoogleBucketLegacyStorageClass.
+        GcsBucketLegacyStorageClass.
```
Comment (jaycarlton) [inline src/main/resources/api/service_openapi.yaml:1323]: I think this type is gone now, since we're not supporting legacy buckets.

**Review by jaycarlton — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (ddietterich) [reply build.gradle:152]: If only :(

File: src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java (hunk 2 above)
Comment (ddietterich) [reply src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:382]: No, I tried not to do that, but it happened a time or three. Fixed it.

File: src/main/java/bio/terra/workspace/service/iam/model/IamRole.java (hunk 4 above)
Comment (ddietterich) [reply src/main/java/bio/terra/workspace/service/iam/model/IamRole.java:9]: Thanks

File: src/main/resources/api/service_openapi.yaml (hunk 6 above)
Comment (ddietterich) [reply src/main/resources/api/service_openapi.yaml:790]: I didn't do that. Just moved it.

**Review by zloery — APPROVED** [review]
LGTM pending tests, I haven't looked to see how real the failures are

File: src/main/java/bio/terra/workspace/service/iam/model/WsmIamRole.java [hunk 10]
```diff
@@ -5,31 +5,31 @@
...
+public enum WsmIamRole {
+  READER("reader", ApiIamRole.READER),
+  WRITER("writer", ApiIamRole.WRITER),
+  OWNER("owner", ApiIamRole.OWNER);
 
   private final String samRole;
   private final ApiIamRole apiRole;
 
-  IamRole(String samRole, bio.terra.workspace.generated.model.ApiIamRole apiRole) {
+  WsmIamRole(String samRole, bio.terra.workspace.generated.model.ApiIamRole apiRole) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/service/iam/model/WsmIamRole.java:16]: you can drop the qualification, here and on line 21.

File: src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java [hunk 11]
```diff
@@ -26,17 +26,21 @@
...
 import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.mock.mockito.MockBean;
 
+@Tag("unit")
 class ReferenceResourceServiceTest extends BaseUnitTest {
```
Comment (jaycarlton) [inline src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java:44]: *referenced

File: src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java [hunk 12]
```diff
@@ -292,273 +296,300 @@ void testInvalidCast() {
...
-      ReferencedResource byid =
-          referenceResourceService.getReferenceResource(
+      try {
+        logger.info("testEnumerate - enumeration");
+        List<ReferencedResource> daoResources =
+            referenceResourceService.enumerateReferences(workspaceId, 0, 100, USER_REQUEST);
+        logger.info("testEnumerate - got {}", daoResources.size());
+        assertThat(daoResources.size(), equalTo(resources.size()));
+      } finally {
+        for (var resource : resources) {
```
Comment (jaycarlton) [inline src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java:319]: oh neat. even if there's an assertion failure you can still clean up.

File: src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java [hunk 13]
```diff
@@ -292,273 +296,300 @@ void testInvalidCast() {
       WsmResource wsmResource = referenceResource;
       assertThrows(InvalidMetadataException.class, wsmResource::castControlledResource);
     }
-  }
-
-  @Nested
```
Comment (jaycarlton) [inline src/test/java/bio/terra/workspace/service/resource/referenced/ReferenceResourceServiceTest.java]: I'm lost in this file. Were tests reordered? Or were a bunch of them deleted?

**Review by jaycarlton — APPROVED** [review]

**Comment by ddietterich** [comment]
Integration tests failed in a weird way. Looks like connectivity issue with Artifactory. I ran them on my personal env and they passed.

**Pull request merged**